### PR TITLE
AgentMessageFeedback Resource

### PR DESCRIPTION
**Pull request by albandum**
## Description

Create AgentMessageFeedback Model & Resource
Design doc is there: https://www.notion.so/dust-tt/Design-Doc-Feedback-loop-thumbs-up-down-14428599d94180bcaab8dc7672e9ea17

## Risk

None

## Deploy Plan

Run migration 121

### CODE REVIEW

File: front/lib/models/assistant/conversation.ts [hunk 1]
```diff
@@ -331,6 +333,79 @@ AgentMessage.init(
...
+});
+AgentMessageFeedback.belongsTo(AgentMessage, {
+  as: "agentMessage",
+  foreignKey: { name: "agentMessageId", allowNull: false },
+});
+AgentMessageFeedback.belongsTo(User, {
+  as: "user",
+  foreignKey: { name: "userId", allowNull: false },
+});
+
```
Comment (tdraier) [inline front/lib/models/assistant/conversation.ts:408]: I think we also need the opposite relation and enforce the onDelete: 'RESTRICT' (at least on the relation which have a resource like user_resource , and implement the cascade in `UserResource.delete` )

```
User.hasMany(AgentMessageFeedback, {
  onDelete: "RESTRICT",
});
```

File: front/lib/resources/agent_message_feedback_resource.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    model: ModelStatic<AgentMessageFeedback>,
+    blob: Attributes<AgentMessageFeedback>
+  ) {
+    super(AgentMessageFeedback, blob);
+  }
+
+  static async makeNew(
+    blob: CreationAttributes<AgentMessageFeedback>
+  ): Promise<AgentMessageFeedbackResource> {
+    const configuration = await AgentMessageFeedback.create({
```
Comment (tdraier) [inline front/lib/resources/agent_message_feedback_resource.ts:31]: nit: variable names , `feedback` or `agentMessageFeedback`

File: front/lib/resources/agent_message_feedback_resource.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      configuration.get()
+    );
+  }
+
+  static async listByAgentConfigurationId({
+    agentConfigurationId,
+  }: {
+    agentConfigurationId: string;
+  }): Promise<AgentMessageFeedbackResource[] | null> {
+    const configuration = await AgentMessageFeedback.findAll({
```
Comment (tdraier) [inline front/lib/resources/agent_message_feedback_resource.ts:46]: same variable name

File: front/lib/models/assistant/conversation.ts [hunk 4]
```diff
@@ -331,6 +333,79 @@ AgentMessage.init(
...
+    },
+    content: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+  },
+  {
+    modelName: "agent_message_feedback",
+    sequelize: frontSequelize,
+  }
```
Comment (tdraier) [inline front/lib/models/assistant/conversation.ts:393]: It's missing some indices here , as we have fetch by `agentConfigurationId` and  `userId`/ `agentMessageId`

File: front/lib/models/assistant/conversation.ts [hunk 5]
```diff
@@ -331,6 +333,79 @@ AgentMessage.init(
...
+  declare agentConfigurationVersion: number;
+  declare agentConfigurationVersionDate: string;
+  declare agentMessageId: ForeignKey<AgentMessage["id"]>;
+  declare userId: ForeignKey<User["id"]>;
+
+  declare thumbDirection: AgentMessageFeedbackDirection;
+  declare content: string | null;
+}
+
+AgentMessageFeedback.init(
```
Comment (flvndvd) [inline front/lib/models/assistant/conversation.ts:354]: Let's add `workspaceId` as well.

File: front/lib/models/assistant/conversation.ts (hunk 1 above)
Comment (flvndvd) [reply front/lib/models/assistant/conversation.ts:408]: ➕ We don't use the default value "ON DELETE CASCADE" anymore.

File: front/lib/resources/agent_message_feedback_resource.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+      order: [["id", "DESC"]],
+    });
+
+    return configuration.map(
+      (config) =>
+        new AgentMessageFeedbackResource(AgentMessageFeedback, config.get())
+    );
+  }
+
+  static async fetchByUserAndMessageId({
```
Comment (flvndvd) [inline front/lib/resources/agent_message_feedback_resource.ts:59]: Can we use pass the `UserResource` here? Ideally we avoid passing string ids when we have access to the resource.

File: front/lib/resources/agent_message_feedback_resource.ts [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    userId,
+    agentMessageId,
+  }: {
+    userId: string;
+    agentMessageId: string;
+  }): Promise<AgentMessageFeedbackResource | null> {
+    const configuration = await AgentMessageFeedback.findOne({
```
Comment (flvndvd) [inline front/lib/resources/agent_message_feedback_resource.ts:66]:
```suggestion
    const agentFeedback = await AgentMessageFeedback.findOne({
```

File: front/lib/resources/agent_message_feedback_resource.ts [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+      AgentMessageFeedback,
+      configuration.get()
+    );
+  }
+
+  async updateContent({
+    content,
+  }: {
+    content: string;
+  }): Promise<Result<undefined, Error>> {
```
Comment (flvndvd) [inline front/lib/resources/agent_message_feedback_resource.ts:87]:
```suggestion
  async updateContent(content: string): Promise<Result<undefined, Error>> {
```

File: front/lib/resources/agent_message_feedback_resource.ts [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+          },
+        }
+      );
+
+      return new Ok(undefined);
+    } catch (error) {
+      return new Err(error as Error);
+    }
+  }
+  async delete(
```
Comment (flvndvd) [inline front/lib/resources/agent_message_feedback_resource.ts:105]:
```suggestion

  async delete(
```

File: types/src/front/assistant/conversation.ts [hunk 10]
```diff
@@ -127,6 +127,8 @@ export type AgentMessageStatus =
   | "failed"
   | "cancelled";
 
+export type AgentMessageFeedbackDirection = "up" | "down";
```
Comment (flvndvd) [inline types/src/front/assistant/conversation.ts:130]: Let's not add those to `types` as we plan to remove them. You can create a file in `lib/api`.

File: front/lib/resources/agent_message_feedback_resource.ts [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+    });
+
+    return new AgentMessageFeedbackResource(
+      AgentMessageFeedback,
+      agentMessageFeedback.get()
+    );
+  }
+
+  static async listByAgentConfigurationId({
+    agentConfigurationId,
```
Comment (spolu) [inline front/lib/resources/agent_message_feedback_resource.ts:43]: Ideally we take an AgentConfigurationType

File: front/lib/resources/agent_message_feedback_resource.ts [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+        new AgentMessageFeedbackResource(AgentMessageFeedback, feedback.get())
+    );
+  }
+
+  static async fetchByUserAndMessageId({
+    user,
+    agentMessageId,
+  }: {
+    user: UserResource;
+    agentMessageId: string;
```
Comment (spolu) [inline front/lib/resources/agent_message_feedback_resource.ts:65]: Same here ideally an AgentMessageType

**Review by spolu — APPROVED** [review]
LGTM

**Pull request merged**